File: News_Service/src/main/java/com/dunice/projectNews/repository/UserRepository.java
package com.dunice.projectNews.repository;

import com.dunice.projectNews.models.User;
import org.springframework.data.jpa.repository.JpaRepository;
import java.util.Optional;
import java.util.UUID;

public interface UserRepository extends JpaRepository<User, UUID> {
    Optional<User> findByEmail(String email);

    void deleteByEmail(String email);

    Boolean existsByEmail(String email);
}

File: News_Service_Kotlin/src/main/kotlin/com/dunice/advancedServiceKotlin/repository/UserRepository.kt
package com.dunice.advancedServiceKotlin.repository

import com.dunice.advancedServiceKotlin.models.User
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository
import java.util.*
@Repository
interface UserRepository : JpaRepository<User, UUID> {
    fun findByEmail(email: String): User

    fun deleteByEmail(email: String)

    fun existsByEmail(email: String): Boolean
}
